### PR TITLE
Add a test for AddAssign on String

### LINKED ISSUE #35047: Add a test for AddAssign on String
https://github.com/rust-lang/rust/pull/34890 was merged, but it should have had an accompanying test.

### PR DESCRIPTION
**Pull request by knight42**
Fix #35047

### CODE REVIEW

**Comment by knight42** [comment]
@aturon  I guess the failure is irrelated to my patch. Maybe run the tests again?

File: src/libcollectionstest/string.rs [hunk 1]
```diff
@@ -192,6 +192,18 @@ fn test_push_str() {
     assert_eq!(&s[0..], "abcประเทศไทย中华Việt Nam");
 }
 
+#[cfg(feature = "stringaddassign")]
```
Comment (durka) [inline src/libcollectionstest/string.rs]: why the cfg?

Comment (knight42) [inline src/libcollectionstest/string.rs]: removed now

**Comment by knight42** [comment]
Could this be merged now?

**Comment by aturon** [comment]
Looks great, thanks!

@bors: r+ rollup

**Comment by durka** [comment]
bors somehow didn't hear that?

**Comment by knight42** [comment]
Perhaps we need retry?

**Pull request merged**